### DISCUSSION TITLE
Issues with Nested Array of Objects

### DISCUSSION THREAD

**Discussion by SemAiSoft** [ngrx/platform#4562] [Q&A]
### Which @ngrx/* package(s) are the source of the bug?

entity, signals, store

### Minimal reproduction of the bug/regression with instructions

I'm a newbie to the world of state mangmt libs, Pls direct with examples on how to PatchState of nested array of objects
For example,

 ```
{
      id: '0',
      children: [
        {
          id: '1',
          children: [
            { id: '3', children: [] },
            { 
                id: '4',
                children: [
                    {
                        id: '7',
                        children: [
                            {
                                id: '8'
                            }
                        ]
                    }
                ] 
            },
          ],
        },
        {
          id: '2',
          children: [
            { id: '5', children: [] },
            { id: '6', children: [] },
          ],
        },
      ],
    };
```

### Expected behavior

Please, how do I find, remove, and patch state of any the 'children' fully or partially, in a non-anti-pattern way.

### Versions of NgRx, Angular, Node, affected browser(s) and operating system(s)

18

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

**Comment by rainerhahnekamp** [comment]
@SemAiSoft, I would advise to flatten the state and create the tree structure via a computed. You would end up with a state type of

```typescript
signalStore(
  withState(Array<{id: number, parentId: number}>)
);
```

Patching should be then much easier.

**Reply by twittwer** [reply]
It may be even easier with a record like `withEntities` is doing it, or am I missing something?

**Reply by SemAiSoft** [reply]
Please, do you mind providing an example on such?

**Reply by twittwer** [reply]
Something like this or use the NgRx/signals' entities package directly 
```ts
signalStore(
  withState({
    entities: Record</*id*/ number, { parentId?: number }>
  })
);
```
